src/ORM/DataList.php

```diff
@@ -625,6 +626,43 @@ public function exclude()
         });
     }
 
+    /**
+     * Return a copy of this list which does not contain any items with any of these params
+     *
+     * @example $list = $list->excludeAny('Name', 'bob'); // exclude bob from list
+     * @example $list = $list->excludeAny('Name', array('aziz', 'bob'); // exclude aziz and bob from list
+     * @example $list = $list->excludeAny(array('Name'=>'bob, 'Age'=>21)); // exclude bob or Age 21
+     * @example $list = $list->excludeAny(array('Name'=>'bob, 'Age'=>array(21, 43))); // exclude bob or Age 21 or 43
+     * @example $list = $list->excludeAny(array('Name'=>array('bob','phil'), 'Age'=>array(21, 43)));
+     *          // bob, phil, 21 or 43 would be excluded
+     *
+     * @param string|array
+     * @param string [optional]
+     *
+     * @return $this
+     */
+    public function excludeAny()
+    {
+        $numberFuncArgs = count(func_get_args());
+        $whereArguments = array();
+
+        if ($numberFuncArgs == 1 && is_array(func_get_arg(0))) {
+            $whereArguments = func_get_arg(0);
+        } elseif ($numberFuncArgs == 2) {
+            $whereArguments[func_get_arg(0)] = func_get_arg(1);
+        } else {
+            throw new InvalidArgumentException('Incorrect number of arguments passed to excludeAny()');
+        }
+        $list = $this;
+
+        foreach ($whereArguments as $field => $value) {
+            $filter = $this->createSearchFilter($field, $value);
+            $list = $list->alterDataQuery(array($filter, 'exclude'));
```

Comment:
Each alterDataQuery creates a new object each time, for performance reason invert this (alterDataQuery on the outside, foreach() on the inside).

Reply:
If you use a closure `function ($query) use ($filter) { /* loop */ $filter->exclude($query); /* end loop */ }` instead of an array `[$filter , 'exclude']` then we can infer usage from other parts of the code.